Sources/TuistCache/Utilities/CacheXCFrameworkBuilder.swift

```diff
@@ -51,16 +58,24 @@ public final class CacheXCFrameworkBuilder: CacheArtifactBuilding {
             }
 
             // Build for the device - if required
-            let deviceArchivePath = temporaryDirectory.appending(component: "device.xcarchive")
-            try await self.deviceBuild(
-                projectTarget: projectTarget,
-                scheme: scheme.name,
-                platform: platform,
-                configuration: configuration,
-                archivePath: deviceArchivePath
-            )
-
-            let productNames = deviceArchivePath
+            var deviceArchivePath: AbsolutePath?
+            let buildForDevice = self.cacheOutputType.shouldBuildForDevice
+            if buildForDevice {
+                deviceArchivePath = temporaryDirectory.appending(component: "device.xcarchive")
+                try await self.deviceBuild(
+                    projectTarget: projectTarget,
+                    scheme: scheme.name,
+                    platform: platform,
+                    configuration: configuration,
+                    archivePath: deviceArchivePath!
+                )
+            }
+
+            guard let archivePath = deviceArchivePath ?? simulatorArchivePath else {
+                return
+            }
```

Comment:
nit: I feel it would be easier to read by getting rid of most of the optionals. This way it's imho easier to follow the code. Hopefully this compiles 😁 

```suggestion
            let archivePath: AbsolutePath
            let deviceArchivePath: AbsolutePath?
            let buildForDevice = self.cacheOutputType.shouldBuildForDevice
            if buildForDevice {
                deviceArchivePath = temporaryDirectory.appending(component: "device.xcarchive")
                try await self.deviceBuild(
                    projectTarget: projectTarget,
                    scheme: scheme.name,
                    platform: platform,
                    configuration: configuration,
                    archivePath: deviceArchivePath
                )
            } else if let simulatorArchivePath = simulatorArchivePath {
                archivePath = simulatorArchivePath
                deviceArchivePath = nil
            } else {
                return
            }
```